src/checkbox.jsx

```diff
@@ -55,7 +55,10 @@ const Checkbox = React.createClass({
   //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: newMuiTheme,
+      switched: nextProps.hasOwnProperty('checked') && nextProps.checked !== this.state.switched,
```

Comment:
Hmm you have a good point, I was just following the previous one. Alright, I'll fix it